src/nnet3bin/nnet3-compute-batch-from-wavs.cc

```diff
@@ -0,0 +1,205 @@
+// nnet3bin/nnet3-compute-batch-from-wavs.cc
+
+// Copyright 2012-2018   Johns Hopkins University (author: Daniel Povey)
+//           2021        Yan Zhao
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "base/timer.h"
+#include "nnet3/nnet-utils.h"
+#include "matrix/kaldi-vector.h"
+#include "nnet3/decodable-batch-looped.h"
+#include "online2/online-nnet2-feature-pipeline.h"
+
+using namespace kaldi;
+using namespace kaldi::nnet3;
+typedef kaldi::int32 int32;
+typedef kaldi::int64 int64;
+
+typedef struct _ThreadPara {
+  std::vector<std::string> wavs_file_name;
+  std::vector<std::string> utts;
+  BaseFloatMatrixWriter    *matrix_writer;
+  std::mutex               mtx;
+  int                      index;
+  
+  OnlineNnet2FeaturePipelineInfo *pipeline_info;
+  NnetBatchLoopedComputer        *computer;
+  TransitionModel                *trans_model;
+
+} ThreadPara; 
+
+
+void *ThreadFunction(void *para) {
+  ThreadPara *thread_para = reinterpret_cast<ThreadPara *>(para);
+  int output_dim = thread_para->computer->GetInfo().output_dim;
+  static int package_size = 1024;
```

Comment:
```suggestion
  constexpr int kPackageSize = 1024;
```

Using `std::array` is even better.

Reply:
ok